Fixes for FATES-HYDRO model

xuchongang opened the pull request:
This pull request updates the following:
1. Fixes the dimension mismatch bug when coupling ELM to FATES-HYDRO 
2. Updates the initial soil water content for FATES-HYDRO; and 
3. Returns to ELM the water stored in the plant when FATES-HYDRO is active

Fixes #2346 

[BFB]

bishtgautam commented on components/clm/src/biogeophys/WaterStateType.F90:
```diff
@@ -670,7 +670,7 @@ subroutine InitCold(this, bounds, &
                      this%h2osoi_vol_col(c,j) = 0.0_r8
                   else
-                     this%h2osoi_vol_col(c,j) = 0.15_r8
+                     this%h2osoi_vol_col(c,j) = 0.70*watsat_col(c,j)  !0.15_r8 to avoid very dry conditions that cause errors in FATES HYDRO
```
I suggest initializating soil moisture to a different value only when FATES HYDRO is active via:

```
if (use_fates_planthydro) then
   this%h2osoi_vol_col(c,j) = 0.70_r8*watsat_col(c,j) 
else
   this%h2osoi_vol_col(c,j) = 0.15_r8
endif
```

xuchongang commented:
@rljacob and @bishtgautam , I have resolved all your review suggestions. Specifically, I have removed my last commit and rebased to the master instead of merging from master. I have opened an issue (2346 to record the bug). I have also made the suggested changes by Gautam. Please let me know if you have further concerns.

bishtgautam commented:
@xuchongang, Thanks for the changes. I have updated the first comment to follow [E3SM's template](https://acme-climate.atlassian.net/wiki/spaces/Docs/pages/16253965/Commit+message+template).
